adaptive_memory_v4.0.py

```
@@ -0,0 +1,2038 @@
+    async def _remove_duplicate_memories(self, user_id: str) -> int:
+        """Remove duplicate memories for a user and return count of removed duplicates."""
+        try:
+            all_memories = Memories.get_memories_by_user_id(user_id)
+            if not all_memories or len(all_memories) < 2:
+                return 0
+                
+            duplicates_removed = 0
+            pipeline = MemoryPipeline(self.valves, self.embedding_manager, self.error_manager)
+            
+            # Check each memory against all others
+            for i, memory in enumerate(all_memories):
```

Ruff (0.14.10)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)